mod_login improvement

cpfeifer opened the pull request:
Pull Request for Issue # 
No Issue reported, suggested through JUX user research.
#### Summary of Changes

Currently there are no options to hide the username and password reminder on the front end. File override is the only option. Two respective buttons have been added to the mod_login admin, with 4 new language strings respectively, and 2 conditionals in default.php to show or hide these two links. 
#### Testing Instructions

Go to a login module, the two new controls are at the bottom of the basic params screen. They are turned on by default as normal, hide the items and check the front end.

![mod_login-pr](https://cloud.githubusercontent.com/assets/3790835/14908806/3b01b944-0d99-11e6-8a5c-982906c93749.jpg)

cpfeifer committed:
Update mod_login.xml

Adding controls to show or hide the username and password reminders on the front end. 2 new yes / no buttons for each respective option. New language strings to be added:
MOD_LOGIN_USERNAMEREMIND_LABEL
Username Reminder
MOD_LOGIN_USERNAMEREMIND_DESC
Show Username Reminder

MOD_LOGIN_PASSWORDREMIND_LABEL
Password Reminder
MOD_LOGIN_PASSWORDREMIND_DESC
Show Password Reminder

cpfeifer committed:
Update default.php

Added conditionals for show / hide the Username and Password reminders respectively. Backend params added in mod_login.xml

cpfeifer committed:
Update en-GB.mod_login.ini

Added new Language Strings for show / hide username and password controls
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_LABEL="Username Reminder"
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_DESC="Show Username Reminder"
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_PASSWORD_LABEL="Password Reminder"
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_PASSWORD_DESC="Show Password Reminder"

cpfeifer committed:
Update mod_login.xml

Updated new language strings to better match existing format, also added in mod_login En-gb language file
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_LABEL="Username Reminder"
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_DESC="Show Username Reminder"
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_PASSWORD_LABEL="Password Reminder"
MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_PASSWORD_DESC="Show Password Reminder"

brianteeman commented:
Please update to Username not User Name. It looks wrong to use both

Bakual commented on modules/mod_login/tmpl/default.php:
```diff
@@ -102,14 +102,18 @@
 				</li>
 			<?php endif; ?>
+			<?php if ($params->get('usernameremind')) : ?>
 				<li>
```
You need to reflect the default values here. So use `$params->get('usernameremind', 1)`. Otherwise the links will vanish after the update and reappear after the user saved the module again, without doing anythign else.
Same for the other parameter below.

brianteeman replied:
Is the aim of this PR purely cosmetic or is it to disable the ability to
reset the Username or password?
If it is cosmetic then I believe an override is the correct route.
If it is to disable the reset then it fails

brianteeman replied:
Also you need to update the override for the Beez template

andrepereiradasilva commented:
For cinsistency, also needs a similiar change in com_users frontend  login view

In https://github.com/joomla/joomla-cms/blob/staging/components/com_users/views/login/tmpl/default_login.php#L89-L96

cpfeifer commented:
Thank you for the comments, this is my first PR :) I'll clean this up tomorrow. 

This is a cosmetic change, meant to simply hide or show these options on the front end, not disable the functionality. A number of users, including myself, have noted that the only reason they ever need to override this view file is to remove these lines of code. It's not always desirable to have these options displayed and I think it makes sense to provide a way to let people choose for themselves what is shown.

SharkyKZ commented:
Perhaps also add an option to hide registration for consistency?

brianteeman commented:
@SharkyKZ you can already do that by disabling the ability for user registration

And that is the problem with this PR and why I am not currently in favour of it. If I had a site where I did not want users to be able to change their username or password then I would expect that setting this new option would achieve that just as disable user registration option does. But it doesnt. I could still change/request the username/password directly using com_users which would mean that I had built a site that I had thought people couldnt change their usernames but they could. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10126">issues.joomla.org/joomla-cms/10126</a>.</sub>

andrepereiradasilva commented:
I agree with @brianteeman i also thing this should be in users component options, not in mod_login

sanek4life commented:
maybe something like this?

we live in era with Emoji and font icons
![pass](https://cloud.githubusercontent.com/assets/3662519/14915474/fa5cf9ac-0e1b-11e6-841e-f1e1b67bf6a2.png)

sanek4life commented:
I mean, you have to put a "question icon" mark for the field user name and password

brianteeman commented:
You mean exactly like the admin login? ;)

On 29 April 2016 at 13:14, s <notifications@github.com> wrote:

> I mean, you have to put a "question icon" mark for the field user name and
> password
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10126#issuecomment-215695795>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

sanek4life commented:
@brianteeman yes)

cpfeifer commented:
I agree the functionality for the registration needs to be handled by the component, changing this functionality is outside the scope of any module, and it's not the goal of this PR.

People want to have control over displaying these items. There are buttons to show and hide everything else in Joomla, including cosmetic details in this specific module. Why force our users to do overrides or write hacky CSS when we can add a simple button to help them do this? I'm thinking about our users and there is plenty of evidence this is something people want. 

**There are many tutorials like this out there:**
https://www.ostraining.com/blog/joomla/remove-links-from-login/
That's a lot of steps.

**These are comments from our users:**
"it might be cool to be able to disable the "Forgot your username" and "Forgot your password" feature in the Login Module, or Options from the User Manager. I'm currently commenting out this bit of code in a mod_login override."

"I ended up using some CSS on clients to avoid overrides..."

"This has probably been asked a thousand times, but i can't seem to find the best answer to remove the 
"forgot username" and "forgot password" links underneath the login box .
How do you remove them both, i only want to see the login box.
if it involves editing the code somewhere, can someone eplain clearly waht to either remove or edit out as i am relatively newbie to joomla."

cpfeifer committed:
Update mod_login.xml

Updated yes / no buttons to show / hide

cpfeifer committed:
Update en-GB.mod_login.ini

Update text label and description for the account reminder links so the intention is more clear.

cpfeifer committed:
Update default.php

Added conditionals for showing or hiding the username and password reminder links

cpfeifer commented:
I've updated everything so it's bit more clear. 

I'll be honest, I've been using Joomla for years and I've never even noticed those ? reminders in the admin login until they were pointed out in this thread. 

In my opinion, adding those icons in would essentially achieve the same thing I'm trying to do, only because not many people know what they are for. Icons and Emoticons are only useful if they effectively communicate the feature and I would say they do not in this case. Too many features are hidden away from our users already.

andrepereiradasilva commented:
what?? that "?" in admin is recover password???

in years with problably thousands admin logins, never ever notice that. tought it was an help tip!

Is it just me or it doesn't make sense?

brianteeman commented:
I see a question mark and I always try to see what it means

On 29 April 2016 at 22:12, andrepereiradasilva <notifications@github.com>
wrote:

> what?? that "?" in admin is recover password???
>
> in years with problably thousands admin logins, never ever notice that.
> tought it was an help tip!
>
> Is it just me or it doesn't make sense?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/10126#issuecomment-215883053>
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

andrepereiradasilva commented:
well, you're more curious than me :smile:

brianteeman commented:
I have tested this item :red_circle: unsuccessfully on 86568b467bce1913e535430bbdd978dbfb53fb19<br /><br />I applied the patch and the two remember links were removed from my site without me doing anything else - obviously changing the default behaviour like this is not correct<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10126">issues.joomla.org/joomla-cms/10126</a>.</sub>

brianteeman commented:
Note that after testing it above I opened the module to see that both settings were set to show<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10126">issues.joomla.org/joomla-cms/10126</a>.</sub>

brianteeman commented on language/en-GB/en-GB.mod_login.ini:
```diff
@@ -20,6 +20,10 @@ MOD_LOGIN_FIELD_USESECURE_DESC="Submit encrypted login data using HTTPS (encrypt
 MOD_LOGIN_FIELD_USESECURE_LABEL="Encrypt Login Form"
 MOD_LOGIN_FIELD_USETEXT_DESC="Choose text or icons to display the field labels. Default is icons."
 MOD_LOGIN_FIELD_USETEXT_LABEL="Display Labels"
+MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_LABEL="Display username reminder"
```
Please capitalise like
Display Username Reminder

brianteeman commented on language/en-GB/en-GB.mod_login.ini:
```diff
@@ -20,6 +20,10 @@ MOD_LOGIN_FIELD_USESECURE_DESC="Submit encrypted login data using HTTPS (encrypt
 MOD_LOGIN_FIELD_USETEXT_LABEL="Display Labels"
+MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_LABEL="Display username reminder"
+MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_USERNAME_DESC="Show or hide the Forgot your username? link"
+MOD_LOGIN_FIELD_SHOW_FORGOT_YOUR_PASSWORD_LABEL="Display password reminder"
```
Please capitalise like
Display Password Reminder

brianteeman commented:
Seems you have top open and then save the login module for this change not to break the existing site settings<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10126">issues.joomla.org/joomla-cms/10126</a>.</sub>

brianteeman commented:
I still believe that this is the wrong approach to a valid issue and it should be done in the User options and not just in the view of the module

![screen shot 2016-04-29 at 16 33 01](https://issues.joomla.org/uploads/1/77e8ea7aef325c526d5dd2bef6517a50.png)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10126">issues.joomla.org/joomla-cms/10126</a>.</sub>

cpfeifer commented:
I think that's a valid point @brianteeman - I would say that many users do not know those options exist in the component, I myself hadn't even considered those specific buttons. What if we put in a message or a link to the component options from the module to point out the functionality already exists?

Many people only consider what they see right in front of them, and my only goal is to make this easier for users. I think we could do that simply by bringing attention to the options they have available. I'm open to anything that helps our users do what they need to do faster and easier.

brianteeman commented:
They don't exist.

That was a mock-up of how I believe this should be done in the correct way.
Which would ensure that the options are DISABLED and not just cosmetically
HIDDEN.

brianteeman commented:
To repeat. Not only does this proposed PR not work it only hides something
that a user will be expecting to disable if they were presented with the
option as it is proposed. This is why disabling registration is not done by
hiding the link but by correctly disabling it.

cpfeifer commented:
I see, that's probably why I've never noticed those options before :)

I think you're right, that seems like a much better approach in terms of functionality. If we can add these options in the appropriate location and then point out they exist, through a message or link in the module admin, I think that helps our users enough and doesn't compromise existing functionality. I'll go back to the drawing board.

Kubik-Rubik commented:
@cpfeifer Thank you, Cliff. It is a good first PR but I agree with @brianteeman here. IMO it makes more sense to move this to the settings of the component and disable these features completely instead of just hiding the links. What you are trying to achieve can be done easily with a template override, I'm not sure whether we need more parameters in the module. I you would like to try the suggested approach, then just ask me if you need any help!

cpfeifer commented:
Thank you @Kubik-Rubik and @brianteeman - I do want to help our users, but only with solutions that make sense for all of us and maintain the integrity of our platform. I'm just getting started so I figured there would be some details I hadn't considered. I appreciate the feedback and the help. I'll dig into this alternate solution and see what we can do. Thanks again.

brianteeman commented:
Let me know if you need any help going down the component option route? It
should be relatively simple if you just follow the example of the disable
registration option

cpfeifer commented:
I appreciate that, I might need it. Let me give it shot and see how it goes over the weekend and I'll do a new PR :)
